File: app/src/main/java/djordjeh/architecture/mvp/dagger/ApplicationModule.java
package djordjeh.architecture.mvp.dagger;

import android.app.Application;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import djordjeh.architecture.mvp.ui.SchedulersFacade;
import djordjeh.architecture.mvp.ui.SchedulersFacadeImpl;

@Module
class ApplicationModule {

    @Provides
    @Singleton
    Context provideContext(Application application) {
        return application.getApplicationContext();
    }

    @Provides
    @Singleton
    SchedulersFacade provideSchedulersFacade() {
        return new SchedulersFacadeImpl();
    }
}

File: app/src/main/java/djordjeh/architecture/mvp/dagger/ApplicationModule.kt
package djordjeh.architecture.mvp.dagger

import android.app.Application
import android.content.Context

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import djordjeh.architecture.mvp.ui.SchedulersFacade
import djordjeh.architecture.mvp.ui.SchedulersFacadeImpl

@Module
internal class ApplicationModule {

    @Provides
    @Singleton
    fun provideContext(application: Application): Context {
        return application.applicationContext
    }

    @Provides
    @Singleton
    fun provideSchedulersFacade(): SchedulersFacade {
        return SchedulersFacadeImpl()
    }
}
